mlflow/pytorch/pytorch_autolog.py

```diff
@@ -0,0 +1,197 @@
+import gorilla
+import logging
+import mlflow.pytorch
+import os
+import pytorch_lightning as pl
+import shutil
+import tempfile
+from pytorch_lightning.core.memory import ModelSummary
+from mlflow.utils.autologging_utils import try_mlflow_log
+
+logging.basicConfig(level=logging.ERROR)
+
+every_n_iter = 1
+
+
+# autolog module uses `try_mlflow_log` - mlflow utility to log param/metrics/artifacts into mlflow
+# Eventhough the same can be achieved using
+# MlflowLogger(Pytorch Lightning's inbuild class),
+# following are the downsides in using MlflowLogger.
+# 1. MlflowLogger doesn't provide a mechanism to dump an entire model into mlflow.
+#    Only model checkpoint is saved.
+# 2. For dumping the model into mlflow `mlflow.pytorch` library is used
+# and the library expects `mlflow` object to be instantiated.
+# In case of MlflowLogger, Run management is completely controlled by the class and
+# hence mlflow object needs to be reinstantiated by setting
+# tracking uri, experiment_id and run_id which may lead to a race condition.
+
+
+def autolog(log_every_n_iter=1):
+    global every_n_iter
+    every_n_iter = log_every_n_iter
+
+    class __MLflowPLCallback(pl.Callback):
+        """
+        Callback for auto-logging metrics and parameters.
+        """
+
+        def __init__(self):
+            self.early_stopping = False
+
+        def on_epoch_end(self, trainer, pl_module):
+            """
+            Log loss and other metrics values after each epoch
+
+            :param trainer: pytorch lightning model instance
+            :param pl_module: pytorch lightning base module
+            """
+            if (pl_module.current_epoch + 1) % every_n_iter == 0:
+                for key, value in trainer.callback_metrics.items():
+                    try_mlflow_log(
+                        mlflow.log_metric, key, float(value + 1), step=pl_module.current_epoch
```

Comment:
Does `value` represent a metric value here?

Reply:
removed the increment.